### DISCUSSION TITLE
Towns and Regions?

### DISCUSSION THREAD

**Discussion by jeff-stienstra** [gadhagod/Hyrule-Compendium-API#29] [Ideas]
Hello,
Excellent job putting this together! I am learning React and needed a fun topic to keep me interested in my own little project, which is how I came across your API. I'm planning out a basic UI to display your API info. I'd love to incorporate it on a map too, but we'll see how it goes.

Have you considered adding all the towns and regions as two additional categories with some basic info and an image? They play such a key role in the storyline and there aren't very many of them. :)

Anyhow, thanks for making this!!

**Town or Settlement Name | Region**
Akkala Ancient Tech Lab | Akkala
Flight Range | Tabantha
Gerudo Town | Gerudo Wasteland
Goron City | Eldin
Hateno Ancient Tech Lab | Hateno
Hateno Village | Hateno
Kakariko Village | Dueling Peaks
Kara Kara Bazaar | Gerudo Wasteland
Korok Forest | Great Hyrule Forest
Lurelin Village | Faron
Rito Village | Tabantha
Tarrey Town | Akkala
Yiga Clan Hideout | Gerudo Wasteland
Zora's Domain | Lanayru

https://www.ign.com/wikis/the-legend-of-zelda-breath-of-the-wild/Towns_and_Settlements#Towns_and_Settlements

**Comment by gadhagod** [comment]
Great suggestion @jeff-stienstra! I like the idea of incorporating more than just the compendium entries into this API (as discussed in #27 and a few other places). 

What do you think should the properties of these settlements should be? Other than the location, what metadata should each town and region possess? The more data the more useful this API can be.

**Comment by jeff-stienstra** [comment]
Is there an open-source reference wiki you used for all the data you have so far that isn't copyrighted?

How about this for starters? I debated between using the key "region" or "location". My intent was for the value to be the region that the village is found in, rather than a description of its location (ie "East of Dueling Peaks and north of blah blah etc"). Perhaps "directions" could be added in the future? You could also add all the inhabitants, nearby shrines, main quests and side quests, but that's a whole new level of info to hunt down and could wait until all the villages are done. 
```
{
    "data": {
        "villages": [
            {
            "name": "Kakariko Village",
            "region": "Dueling Peaks",
            "description": "Kakariko Village is a bustling, Ancient Japanese-inspired town located in the Dueling Peaks Region. It's most likely the first town you visit. There is also a nearby shrine, known as Ta'loh Naeg Shrine and a mountain passage to the Great Fairy Fountain.",
            "image": "https://oyster.ignimgs.com/mediawiki/apis.ign.com/the-legend-of-zelda-hd/6/6a/Kakariko_Village.jpg?width=1280",
            "merchants": ["High Spirits Produce", "Olkin's Pumpkins", "The Curious Quiver", "Shuteye Inn", "Enchanted - Armor Shop", "Kilton the Monster Parts Merchant"]
            }
        ]
    }
}

**Reply by jeff-stienstra** [reply]
Here is some basic info for regions:
```
"regions": [
            {
            "name": "Akkala",
            "location": "Far Northeastern reaches of Hyrule",
            "description": "The Akkala region is a dangerous area in the far Northeastern reaches of Hyrule. It is a beautiful region with many forests, lakes, and grassy plains - backed by the Eldin mountains on the West, and the ocean shores along the East.",
            "images": {
                "map": "https://cdn.vox-cdn.com/uploads/chorus_asset/file/10071555/Akkala_region_shrine_maps.jpg",
                "photo": "https://cdn.vox-cdn.com/thumbor/fKKdrox05URZNA_Mfqt6_WI_cXw=/99x0:1179x720/1200x800/filters:focal(99x0:1179x720)/cdn.vox-cdn.com/uploads/chorus_image/image/53575943/2017030700165900-F1C11A22FAEE3B82F21B330E1B786A39.0.0.jpg"
                },
            "shrines":["Dah Hesho Shrine", "Katosa Aug Shrine", "Ke'naj Shakah Shrine", "Ritaag Zumo Shrine", "Tu Ka'loh Shrine", "Tutsuwa Nima Shrine", "Ze Kasho Shrine", "Zuna Kai Shrine"]
            }
        ]

**Comment by gadhagod** [comment]
Added to v3 plan: https://github.com/gadhagod/Hyrule-Compendium-API/issues/30.

**Comment by gadhagod** [comment]
https://github.com/gadhagod/Hyrule-Compendium-API/issues/30#issuecomment-1211493489
